apps/io-wallet-user-func/src/wallet-instance.ts

```diff
@@ -9,29 +10,54 @@ import { EntityNotFoundError } from "io-wallet-common/http-response";
 import { JwkPublicKey } from "io-wallet-common/jwk";
 import { User } from "io-wallet-common/user";
 
-class WalletInstanceRevoked extends Error {
+class RevokedWalletInstance extends Error {
   name = "WalletInstanceRevoked";
   constructor() {
     super("The wallet instance has been revoked.");
   }
 }
 
-export const WalletInstance = t.type({
+const WalletInstanceBase = t.type({
+  createdAt: IsoDateFromString,
   hardwareKey: JwkPublicKey,
   id: NonEmptyString,
-  isRevoked: t.boolean,
   signCount: t.number,
   userId: User.props.id,
 });
 
+const WalletInstanceValid = t.intersection([
+  WalletInstanceBase,
+  t.type({
+    isRevoked: t.literal(false),
+  }),
+]);
+
+type WalletInstanceValid = t.TypeOf<typeof WalletInstanceValid>;
+
+const WalletInstanceRevoked = t.intersection([
+  WalletInstanceBase,
+  t.type({
+    isRevoked: t.literal(true),
+    revokedAt: IsoDateFromString,
+  }),
+]);
+
+export const WalletInstance = t.union([
+  WalletInstanceValid,
+  WalletInstanceRevoked,
+]);
+
 export type WalletInstance = t.TypeOf<typeof WalletInstance>;
 
 export interface WalletInstanceRepository {
-  batchPatchWithReplaceOperation: (
-    operations: {
+  batchPatch: (
+    operationsInput: {
       id: string;
-      path: string;
-      value: unknown;
+      operations: {
+        op: "add" | "incr" | "remove" | "replace" | "set";
```

Comment:
Are you sure you need all of these?

Reply:
Ok, it's usually best to leave only the ones you need. Less is more

Reply:
if you want to reduce the operations it would no longer be a patch, so the name should be changed